pkg/sql/logictest/logic.go

```diff
@@ -3028,8 +3028,7 @@ func (t *logicTest) processSubtest(
 				}
 				delete(m, nodeIdx)
 			}
-			db := t.getOrOpenClient(t.user, nodeIdx)
-			t.db = db
+			t.setUser(t.user, nodeIdx)
```

Comment:
do you think we should use 
```
			if t.nodeIdx == nodeIdx {
				t.setUser(t.user, nodeIdx)
			}
```

i could see people getting confused that `upgrade` changes the node you're connected to

Reply:
Yeah, I agree that people might get confused if the node changes after an upgrade. Updated based on your suggestion!